src/WGetTest/Program.cs

```diff
@@ -23,97 +23,135 @@ private void Run()
+                    Console.WriteLine($"Source: {source.Name} ({source.Identifier})");
                 }
 
-                Console.WriteLine(connector.PinAdd("7zip.7zip", true));
-                List<WinGetPinnedPackage> pinnedList1 = connector.GetPinnedPackages();
-                Console.WriteLine(pinnedList1[0].Name + ": " + pinnedList1[0].PinTypeString);
-                Console.WriteLine(connector.PinRemove("7zip.7zip"));
-                Console.WriteLine(connector.PinAdd("7zip.7zip", "23.*"));
-                List<WinGetPinnedPackage> pinnedList2 = connector.GetPinnedPackages();
-                Console.WriteLine(pinnedList2[0].Name + ": " + pinnedList2[0].PinTypeString);
-                Console.WriteLine(connector.PinRemove("7zip.7zip"));
+                bool sourceUpdateStatus = sourceManager.UpdateSources();
+                Console.WriteLine($"Source Update Status: {sourceUpdateStatus}");
 
-                Console.WriteLine(connector.PinAddInstalled("7zip.7zip", true));
-                Console.WriteLine(connector.PinRemoveInstalled("7zip.7zip"));
-                Console.WriteLine(connector.PinAddInstalled("7zip.7zip", "23.*"));
-                Console.WriteLine(connector.PinRemoveInstalled("7zip.7zip"));
+                string json = sourceManager.SourcesToJson(sourceList);
+                Console.WriteLine($"Sources as JSON: {json}");
 
-                Console.WriteLine(connector.ResetPins());
+                sourceManager.ExportSourcesToFile(@"Tests\AllSources.json");
+                sourceManager.ExportSourcesToFile(@"Tests\msstoreSources.json", "msstore");
 
-                Console.WriteLine("Same Package:");
+                // Hash Test
+                Console.WriteLine("\n=== Hash Test ===");
+                string hash = connector.Hash(@"Tests\AllSources.json");
+                Console.WriteLine($"Hash of 'AllSources.json': {hash}");
 
+
+                // Package comparison examples with more information
+                Console.WriteLine("\n=== Package Comparison Tests ===");
                 WinGetPackage p1 = WinGetPackage.Create("SampleP1", "SampleId1", "1.0.0", "SampleSource");
                 WinGetPackage p2 = WinGetPackage.Create("SampleP2", "SampleId2", "2.0.0", "3.0.0", "");
                 WinGetPackage p3 = WinGetPackage.Create("SampleP1", "SampleId1", "2.0.0", "SampleSource");
-
-                Console.WriteLine(p1.SamePackage(p1)); // true
-                Console.WriteLine(p1.SamePackage(p2)); // false
-                Console.WriteLine(p1.SamePackage(p3)); // true
-                Console.WriteLine(p1.SamePackage(p3, true)); // false
-                Console.WriteLine(p2.SamePackage(p2)); // true
-                Console.WriteLine(p2.SamePackage(p1)); // false
-                Console.WriteLine(p2.SamePackage(p3)); // false
+                bool PackageTests = true;
+                PackageTests = PackageTests && p1.Equals(p1);
```

Comment:
These Tests were meant to test the “SamePackage” Method, which is not equal to the “Equals” method, because it takes the package version into consideration or ignores it, depending on the method call.
Therefore the “SamePackage” Method should still be used for this test.
But it might make sense to add an additional equals test block.